src/datarobotx/idp/retraining_policies.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 DataRobot, Inc. and its affiliates.
+#
+# All rights reserved.
+#
+# DataRobot, Inc.
+#
+# This is proprietary source code of DataRobot, Inc. and its
+# affiliates.
+#
+# Released under the terms of DataRobot Tool and Utility Agreement.
+# https://www.datarobot.com/wp-content/uploads/2021/07/DataRobot-Tool-and-Utility-Agreement.pdf
+
+import json
+from typing import Any, Dict, Union
+
+import datarobot as dr
+
+
+def _check_response(response: Any) -> None:
+    if response.status_code < 200 or response.status_code > 399:
+        raise Exception(f"Request failed with status code {response.status_code}: {response.text}")
+
+
+def update_or_create_retraining_policy(
+    endpoint: str,
+    token: str,
+    deployment_id: str,
+    name: str,
+    dataset_id: Union[str, None] = None,
+    **kwargs: Any,
+) -> str:
+    """Update or create a retraining policy for a model deployment.
+
+    Parameters
+    ----------
+    endpoint : str
+        The endpoint for the DataRobot API.
+    token : str
+        The token for authentication with the DataRobot API.
+    deployment_id : str
+        The ID of the deployment to update or create the retraining policy for.
+    name : str
+        The name of the retraining policy.
+    dataset_id : Union[str, None], optional
+        The ID of the dataset used for retraining, by default None.
+    **kwargs : Any
+        Additional keyword arguments for the retraining policy.
+
+    Returns
+    -------
+    str
+        The ID of the created retraining policy.
+    """
+    client = dr.Client(token=token, endpoint=endpoint)  # type: ignore
+
+    # Configure retraining settings
+    if dataset_id is not None:
```

Comment:
This is an interesting one. I think the logic here is solid but good to be mindful of implications: 

1. If a user is creating a first retraining job and doesn't pass in a dataset_id, this call will fail
2. But if a retraining job already exists and a user passes in a dataset_id, this call is ok

We could instead make `dataset_id` a required argument, though I'm unsure how important it is

Reply:
am assuming that the function signature changed - i see it's required. if that's the case this conditional should go away